python/lsst/meas/astrom/denormalizedMatches.py

```diff
@@ -0,0 +1,58 @@
+import lsst.afw.table
+
+__all__ = ["matchesToCatalog"]
+
+def matchesToCatalog(matches, matchMeta=None):
```

Comment:
I find your module name more descriptive than this function name.   How about "denormalizeMatches"?  Or denormalizeRefMatch?or joinReferenceSource? (since in database lingo, I'd say this is creating a materialized view of the joined reference and source cat dimensions.)